packages/dx-grid-core/src/plugins/virtual-table-state/helpers.ts

```diff
@@ -143,7 +143,7 @@ export const needFetchMorePages: PureComputed<[VirtualRows, number, number], boo
   const { start, end } = intervalUtil.getRowsInterval(virtualRows);
   const loadCount = end - start;
   const topTriggerIndex = start > 0 ? start + pageSize! : 0;
-  const bottomTriggerIndex = end - pageSize!;
+  const bottomTriggerIndex = Math.max(topTriggerIndex + pageSize, end - pageSize! * 1.5);
```

Comment:
What is a `1.5` value?

Reply:
Half of page.
Sometimes difference between `bottomTriggerIndex` and `topTriggerIndex` more than one `pageSize` and we can get index, than no trigger update. This is index in interval between `end - pageSize!` and `end - pageSize! * 1.5`
For example, in master try after load press `End` button twice (lazy loading demo)
You will stuck at string 150 as last with `bottomTriggerIndex` === 150 and `index ~ 140`

Reply:
Sometimes we may catch this bug not in beginning of table. It is quite difficult to catch, but can.